lib/spack/spack/version.py

```diff
@@ -1250,10 +1253,7 @@ def __init__(self, pkg_name):
     def cache_key(self):
         if not self._cache_key:
             key_base = "git_metadata"
-            if not self.repository_uri.startswith("/"):
-                key_base += "/"
-            self._cache_key = key_base + self.repository_uri
-
+            self._cache_key = Path(key_base).joinpath(self.repository_uri)
```

Comment:
When using pathlib be sure that none of the `pathlib.Path` objects bleed out from their local useage as much of Spack/the `os` module do not handle those objects well. It looks like you might be returning a `Path` object here.

Reply:
If the path you're constructing does not exist on the filesystem you may also want to consider using `PurePath`

Reply:
We now return the posix path so we don't leak the Pathlib object